assets/src/stories-editor/components/editor-carousel/index.js

```diff
@@ -24,74 +23,87 @@ import './edit.css';
 const TOTAL_PAGE_MARGIN = 50;
 const PAGE_BORDER = 1;
 
-class EditorCarousel extends Component {
-	translateWrapper = () => {
+const EditorCarousel = () => {
+	const { pages, currentPage, currentIndex, previousPage, nextPage, isReordering } = useSelect( ( select ) => {
+		const {
+			getBlockOrder,
+			getBlocksByClientId,
+			getAdjacentBlockClientId,
+		} = select( 'core/block-editor' );
+		const { getCurrentPage, isReordering: _isReordering } = select( 'amp/story' );
+
+		const _currentPage = getCurrentPage();
+		const _pages = getBlocksByClientId( getBlockOrder() );
+
+		const index = pages.findIndex( ( { clientId } ) => clientId === currentPage );
+
+		return {
+			pages: _pages,
+			currentPage: _currentPage,
+			currentIndex: Math.max( 0, index ), // Prevent -1 from being used for calculation.
+			previousPage: getCurrentPage() ? getAdjacentBlockClientId( _currentPage, -1 ) : null,
+			nextPage: getCurrentPage() ? getAdjacentBlockClientId( _currentPage, 1 ) : null,
+			isReordering: _isReordering(),
+		};
+	}, [] );
+
+	useEffect( () => {
 		const wrapper = document.querySelector( '#amp-story-controls + .block-editor-block-list__layout' );
 
 		if ( ! wrapper ) {
 			return;
```

Comment:
Under which circumstances does this return and could this perhaps be triggered by other props then the two used in the guard for the `useEffect`? The old code re-ran on every component update regardless of which prop changed and I could fear that this part here might be related to another prop?

Reply:
Good question. Will have to test this and perhaps search a bit in git history.

Reply:
Did a slight refactor here and removed that condition.